specification/communication/resource-manager/Microsoft.Communication/preview/2023-03-01-preview/SenderUsernames.json

```diff
@@ -0,0 +1,333 @@
+        "tags": [
+          "SenderUsernames"
+        ],
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "Parameters for the create or update operation",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SenderUsernameResource"
+            }
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "EmailServices.json#/parameters/EmailServiceName"
+          },
+          {
+            "$ref": "Domains.json#/parameters/DomainName"
+          },
+          {
+            "$ref": "#/parameters/SenderUsername"
+          }
+        ],
+        "consumes": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "The create or update request was completed successfully",
+            "schema": {
+              "$ref": "#/definitions/SenderUsernameResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-examples": {
+          "Create or update SenderUsernames resource": {
+            "$ref": "./examples/senderUsernames/createOrUpdate.json"
+          }
+        }
+      },
+      "delete": {
+        "description": "Operation to delete a SenderUsernames resource.",
+        "operationId": "SenderUsernames_Delete",
+        "summary": "Delete",
+        "tags": [
+          "SenderUsernames"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "EmailServices.json#/parameters/EmailServiceName"
+          },
+          {
+            "$ref": "Domains.json#/parameters/DomainName"
+          },
+          {
+            "$ref": "#/parameters/SenderUsername"
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
>200 [](http://example.com/codeflow?start=11&length=3)

You should include 204 as a possible response code.

"The resource provider can return 200 (OK) or 204 (NoContent) to indicate that the operation completed successfully. A 200 (OK) should be returned if the object exists and was deleted successfully; and a 204 (NoContent) should be used if the resource does not exist and the request is well formed."

Reply:
Sudipto (@usvoyager ) please confirm we are good with returning the 204 here. Seems like a better experience thatn returning not found